### PR TITLE
CIF-756 - Manually configuration of used GraphQL client/catalog

### PR DESCRIPTION
**Pull request by herzog31**
## Description

* Created datasource servlet to retrieve all available GraphQL clients
* Added select field to CIF page properties to select GraphQL client for the subtree. An empty option `Inherit` is provided that removes the property from the node.
* Added textfield to CIF page properties to specify Magento store view.
* Updated dialog of `catalogpage` component so that the merge of the dialogs works properly.
* Added unit tests for datasource servlet.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/GraphqlClientDataSourceServlet.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    protected List<Resource> getGraphqlClients(@NotNull SlingHttpServletRequest request) {
+        ResourceResolver resolver = request.getResourceResolver();
+        List<Resource> graphqlClients = new ArrayList<>();
+
+        List<Configuration> configs = null;
+        try {
+            configs = Arrays.asList(
+                configurationAdmin.listConfigurations("(service.factoryPid=com.adobe.cq.commerce.graphql.client.impl.GraphqlClientImpl)"));
```
Comment (cjelger) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/GraphqlClientDataSourceServlet.java]: When possible, I think we should avoid hard-coding implementation specific references. You could replace this with something similar to https://github.com/adobe/commerce-cif-connector/blob/master/bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/magento/GraphqlDataServiceImpl.java#L76-L94 where you can only keep a `Set` of identifiers.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/GraphqlClientDataSourceServlet.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        } catch (Exception e) {
+            return graphqlClients;
+        }
+
+        // Add empty option
+        graphqlClients.add(new GraphqlClientResource(i18n.get("Inherit", "Inherit property"), StringUtils.EMPTY, resolver));
+
+        // Add other configurations
+        for (Configuration config : configs) {
+            String identifier = (String) config.getProperties().get("identifier");
```
Comment (cjelger) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/GraphqlClientDataSourceServlet.java]: With the change I propose, you can use `graphqlClient.getdentifier()`

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v1/page/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                                sling:resourceType="granite/ui/components/coral/foundation/form/select"
+                                                fieldDescription="The GraphQL client instance. If this list is empty check the configurations of the CIF GraphQL Client Configuration service. "
+                                                fieldLabel="GraphQL Client"
+                                                name="./cq:graphqlClient"
+                                                required="{Boolean}false">
+                                                <datasource
+                                                    jcr:primaryType="nt:unstructured"
+                                                    sling:resourceType="core/cif/components/page/v1/datasource/graphqlclients"/>
+                                                <granite:data
+                                                        jcr:primaryType="nt:unstructured"
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v1/page/_cq_dialog/.content.xml]: please fix indentation

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v1/page/_cq_dialog/.content.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                                            </graphqlclient>
+                                            <storeview
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                                fieldDescription="Identifier to select Magento Store View. If empty, the default store will be used."
+                                                fieldLabel="Store View"
+                                                name="./cq:magentoStore"
+                                                renderReadOnly="{Boolean}false">
+                                                <granite:data
+                                                        jcr:primaryType="nt:unstructured"
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v1/page/_cq_dialog/.content.xml]: please fix indentation

File: bundles/core/pom.xml [hunk 5]
```diff
@@ -495,6 +495,11 @@
             <version>3.0.0-magento232</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>com.intellij</groupId>
```
Comment (dplaton) [inline bundles/core/pom.xml]: Why do we need this dependency?

Comment (herzog31) [reply bundles/core/pom.xml]: This one is required for the `@NotNull` annotation. The same the WCM datasource servlets use as well. For reference, see https://github.com/adobe/aem-core-wcm-components/blob/development/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AllowedHeadingElementsDataSourceServlet.java#L31.

Comment (dplaton) [reply bundles/core/pom.xml]: I figured it out eventually, but we should use the same annotations from `javax.annotation` (e.g. `javax.annotation.Nonnull`). Introducing a dependency on a vendor library for some trivial annotations seems wrong to me.

Comment (herzog31) [reply bundles/core/pom.xml]: Done :)

**Pull request merged**